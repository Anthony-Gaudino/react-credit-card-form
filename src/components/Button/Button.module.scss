@use "../../theme.scss";

.button {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 1rem;
  font-size: 1rem;
  color: theme.$white;
  cursor: pointer;
  background: theme.$primary-color;
  border: none;
  border-radius: 8px;
  box-shadow: 3px 10px 20px 0 rgb(35 100 210 / 30%);

  &:hover {
    background: theme.$hover-color;
  }

  &--full-width {
    width: 100%;
  }
}
