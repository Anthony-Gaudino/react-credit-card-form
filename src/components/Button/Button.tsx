import clsx from "clsx";
import { MouseEventHandler } from "react";

import styles from "./Button.module.scss";

interface ButtonProps {
  /**
   * If `true` the button will take the full width fo it's parent.
   */
  fullWidth?: boolean;
  /**
   * Button children.
   */
  children: React.ReactNode;
  /**
   * Button onClick callback.
   */
  onClick?: MouseEventHandler;
}

/**
 * A HTML button.
 */
export const Button = (props: ButtonProps): JSX.Element => {
  const { fullWidth = false, children, ...others } = props;

  const classes = clsx({
    [styles.button]: true,
    [styles["button--full-width"]]: fullWidth,
  });

  return (
    <button className={classes} {...others}>
      {children}
    </button>
  );
};
