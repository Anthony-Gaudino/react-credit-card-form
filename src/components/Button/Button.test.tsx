import { fireEvent, render, screen } from "@testing-library/react";

import Button from "../Button";

let clicked = false;
const clickHandler = () => (clicked = true);

describe("Test Button component", () => {
  test("the button is correctly rendered", () => {
    const { asFragment } = render(<Button>Submit</Button>);

    const buttonText = screen.getByText("Submit");

    expect(buttonText).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("handles click events", async () => {
    render(<Button onClick={clickHandler}>Submit</Button>);
    const button = screen.getByRole("button");

    await fireEvent.click(button);

    expect(clicked).toEqual(true);
  });
});
