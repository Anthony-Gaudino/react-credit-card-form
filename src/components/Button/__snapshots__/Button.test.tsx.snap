// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component the button is correctly rendered 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Submit
  </button>
</DocumentFragment>
`;
